### PR TITLE
feat: add block controls

### PR DESCRIPTION
**Pull request by karlpawlowicz**
## Purpose

1. Added up and down controls to blocks, so blocks can be moved. These are only present on blocks that accept other blocks right now. I will need to go back and update the styling of the other blocks, so they can have these and future controls such as a delete and options.
2. Added the start of color highlighting when dropping a block.

## Testing Instructions

1. Confirm the up control moves a button up.
2. Confirm the down control moves a button down.
3. Confirm the main drop zone highlights light green when a block is dragged over it.
4. Confirm blocks that accept other blocks highlight light green when a block is dragged over it.

## Checklist

- [x] Title follows the [naming convention](../README.md#pull-requests).
- [x] Add a description.
- [ ] Add screenshots (optional).
- [x] Add testing instructions.
- [x] Add at least one reviewer.
- [x] Apply related labels (optional).
- [x] Leave comments on changes you would like to discuss.
- [x] Remove changes that are not related to this pull request.
- [x] Remove debugging code.
- [x] Test your changes locally and on staging (if applicable).
- [ ] Ensure test coverage value in config matches actual coverage (if applicable).
- [ ] Ensure the build is passing.

### CODE REVIEW

File: components/editor/DragItemWithDropZone.tsx [hunk 1]
```diff
@@ -60,23 +62,79 @@ const DragItemWithDropZone: FC<DragItemWithDropZoneProps> = ({
...
+      type: 'MOVE_ITEM',
+    })
+  }
+
   return (
     <div ref={drag}>
       <div ref={drop}>
         {(() => {
           switch (type) {
             case PrimitiveTypes.PRIMITIVE_CONTAINER:
```
Comment (mattxyzeth) [inline components/editor/DragItemWithDropZone.tsx]: Do we need to add a case for every primitive block we offer?

File: components/editor/primitives/PrimitiveForm.tsx [hunk 2]
```diff
@@ -1,13 +1,63 @@
 import { FC, ReactNode } from 'react'
+import clsx from 'clsx'
+import { ChevronDownIcon, ChevronUpIcon } from '@heroicons/react/20/solid'
+
+const baseStyles: string =
```
Comment (mattxyzeth) [inline components/editor/primitives/PrimitiveForm.tsx]: I think TS can infer a string type in this case

File: components/editor/primitives/PrimitiveContainer.tsx [hunk 3]
```diff
@@ -1,14 +1,62 @@
...
   children?: ReactNode
+  handleDownClick: Function
+  id: any
+  isOver: Boolean
+  showButtons: Boolean
+  handleUpClick: Function
 }
 
-const PrimitiveContainer: FC<Props> = ({ children }) => {
+const PrimitiveContainer: FC<Props> = ({
```
Comment (mattxyzeth) [inline components/editor/primitives/PrimitiveContainer.tsx:23]: Let's consolidate all the `Primitive` components into this one. Add a prop to pass in `options` with properties like name and type. These options will be fetched from the network.

File: context/Reducer.tsx [hunk 4]
```diff
@@ -23,9 +24,51 @@ const addBlockToChild = (element: any, payload: any) => {
   }
 }
 
+const moveChildItem = (element: any, payload: any) => {
+  const childIndex = element.children.findIndex(
+    (child: any) => child.id === payload.id
+  )
+
+  if (childIndex !== -1) {
```
Comment (mattxyzeth) [inline context/Reducer.tsx]: `> -1` is better here, unless it's meant to be specific.

File: context/types.ts [hunk 5]
```diff
@@ -19,6 +19,11 @@ export type GlobalProviderProps = {
   children: ReactNode
 }
 
+export type MoveItemAction = {
+  payload: any
```
Comment (mattxyzeth) [inline context/types.ts]: Is it possible to be more specific here? What are the different types payload can be?

**Review by mattxyzeth — CHANGES_REQUESTED** [review]
Let's consolidate those Primitive components into one.

File: components/editor/DragItemWithDropZone.tsx (hunk 1 above)
Comment (karlpawlowicz) [reply components/editor/DragItemWithDropZone.tsx]: I don't think so, we can consolidate them for now.

**Review by mattxyzeth — APPROVED** [review]
lgtm

**Pull request merged**